src/lib.rs

```diff
@@ -461,7 +464,11 @@ fn full_fence() {
         // x86 platforms is going to optimize this away.
```

Comment:
The fact that you are hoping that "sane" compilers *for particular targets* are going to treat the memory model differently, is a big red flag. The memory model is target-independent, and a whole bunch of optimizations run on this code (including its use of atomics) before any target-specific concerns are applied.

Inline assembly is the *only* correct choice here.